src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcTaskHost.cs

```diff
@@ -455,40 +465,40 @@ internal static string GetMSBuildLocationFromHostContext(HandshakeOptions hostCo
 
                 toolPath = s_pathToX32Clr2;
             }
-            else if ((hostContext & HandshakeOptions.X64) == HandshakeOptions.X64)
+            else if (IsHandshakeOptionEnabled(HandshakeOptions.X64))
             {
-                if (s_pathToX64Clr4 == null)
+                if (IsHandshakeOptionEnabled(HandshakeOptions.NET))
                 {
-                    s_pathToX64Clr4 = s_baseTaskHostPath64;
-                }
+                    var msbuildAssemblyName = Environment.GetEnvironmentVariable("MSBUILD_ASSEMBLY_NAME");
```

Comment:
since you have issues with Netcore trying to launch netcore, this is very suspicious!

in that case in main branch this method takes 		hostContext:	TaskHost | X64 | NET
and returns D:\msbuild\artifacts\bin\bootstrap\core\sdk\9.0.203\MSBuild.dll
because s_pathToX64Clr4 contains the correct path.

It's unfortunate semantics but it works.

now you broke it by forcing TaskHost | X64 | NET into the codepath using the env var which fails if not set